microapps/GeometryEditorApp/app/src/main/java/com/arcgismaps/toolkit/geometryeditorapp/screens/MainScreen.kt

```diff
@@ -0,0 +1,241 @@
+import androidx.compose.material.icons.filled.MoreVert
+import androidx.compose.material3.DropdownMenu
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Switch
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import com.arcgismaps.Color
+import com.arcgismaps.geometry.Geometry
+import com.arcgismaps.geometry.GeometryType
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.symbology.SimpleFillSymbol
+import com.arcgismaps.mapping.symbology.SimpleFillSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.mapping.view.geometryeditor.GeometryEditor
+import com.arcgismaps.mapping.view.geometryeditor.VertexTool
+import com.arcgismaps.toolkit.geocompose.GraphicsOverlayCollection
+import com.arcgismaps.toolkit.geocompose.MapView
+import com.arcgismaps.toolkit.geocompose.rememberGraphicsOverlayCollection
+
+// line symbol of the graphic sketched on the map
+private val lineSymbol: SimpleLineSymbol by lazy {
+    SimpleLineSymbol(
+        style = SimpleLineSymbolStyle.Solid,
+        color = Color.black,
+        width = 2f
+    )
+}
+
+// fill symbol of the graphic sketched on the map
+private val fillSymbol: SimpleFillSymbol by lazy {
+    SimpleFillSymbol(
+        style = SimpleFillSymbolStyle.Cross,
+        color = Color.cyan,
+        outline = lineSymbol
+    )
+}
+
+/**
+ * Displays a composable [MapView] to add graphics with the [GeometryEditor] using [VertexTool]
+ * on a [rememberGraphicsOverlayCollection]. The editing of graphics can be started/stopped using a [Switch].
+ * Each new sketch is added as a new [GraphicsOverlay]. An action button provides a choice of options
+ * to undo, redo, clear sketch or reset all the graphics overlays.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun MainScreen() {
+    val arcGISMap by remember { mutableStateOf(ArcGISMap(BasemapStyle.ArcGISStreets)) }
+    // the collection of graphics overlays used by the MapView
+    val graphicsOverlays = rememberGraphicsOverlayCollection()
+    // the geometry editor used to manage the editing session
+    val geometryEditor by remember { mutableStateOf(GeometryEditor()) }
```

Comment:
This doesn't need to be a mutable state:
```suggestion
    val geometryEditor = remember { GeometryEditor() }
```